### ISSUE TITLE
GameEngine.reset() does not remove all of the gameObjects

### ISSUE THREAD

**Issue by jtara1** [Jester565/SpyNinjaGame#23]
I've added a feature in testing that resets game and returns to main menu ca21807b6cb59a3262f2246988693e7f05244903

This way makes sense to me, but when I test the feature out there's at least twice as many `GameObject`s on the grid than I'd expect there to be.

To solve this I think the following would work:

* reset grid - this could be done in the `Grid` class or the `GameEngine` class
* insanitiate the Spy in the GameEngine.reset() method

Game print to show issue:

```
Pause Menu
1: Resume | 2: Save | 3: Main Menu | 4: Exit Game
3
Welcome to Spy Game (by Random 6)!

Select an option:
1. Start New Game.
2. Quit.
1
New game started! 
[ ][ ][r][ ][N][ ][N][ ][ ]   Lives: 3
[ ][R][ ][ ][R][r][ ][R][N]   Ammo: 0
[ ][N][ ][ ][ ][ ][N][ ][ ]   Invincible Turns: 0
[N][N][ ][N][I][ ][ ][B][ ]   Radar: Disabled
[N][R][ ][N][R][ ][N][R][ ]
[ ][ ][I][N][ ][B][ ][ ][ ]
[ ][I][ ][ ][ ][N][N][ ][ ]
[S][R][r][N][R][ ][B][b][N]
[S][ ][ ][ ][ ][N][N][ ][ ]

Enter a direction to look in or another command
W: Up | A: Left | S: Down | D: Right
2: Debug | 3: More Options
```

**Comment by Jester565** [comment]
Actually, the solution can be a lot easier than making a reset method in the Grid.  Just don't initialize anything in the GameEngine until the reset method is called.  I implement this here: 3df1fd49ce0fad206a270f3d69d606ed462505ab .  Should be an easy fix.

**Comment by jtara1** [comment]
Ran some tests by going to the "Main Menu" through in the game commands and starting a new game again.

Everything appears to be normal as expected from a `reset()` method.

Closing with commit ff2e21a95a0bc3f80957b956786e02302c3bba31

---

edit: forgot to move player then reset game through in-game menu

Hao caught this and fixed it in commits b9db8ae04450631fb3094b1e12922b275cb91cfc  3ac6bbb64836138f956fb5fbf2b15d4e33f1e685
